lazy.md

```diff
@@ -13,3 +13,623 @@ author:
+3. A coroutine task needs to be sender friendly: it is expected that
+    asynchronous code is often written using coroutines awaiting
+    senders. However, it isn't as obvious as not all senders will
+    necessarily be awaitable. For example neither unifex nor stdexec
+    support senders with more than one `set_value` completion.
+4. It is possibly confusing and problematic if coroutines resume on
+    a different execution context than the one they were suspended
+    on.  Senders could, however, complete on an entirely different
+    scheduler than where they started. As a result, a coroutine
+    task should be scheduler affine by default, i.e., it should
+    always resume on the same scheduler. There should probably be
+    a way to opt out of scheduler affinity when the implications are
+    well understood.
+5. When using coroutines there will probably be an allocation at
+    least for the coroutine frame. To support the use in environments
+    where memory allocations using `new`/`delete` aren't supported
+    the coroutine task should support allocations using allocators.
+6. Receivers have associated environments which can support an open
+    set of queries. Normally, queries on an environment can be
+    forwarded to the environment of a `connect()`ed receiver.
+    Since the coroutine types are determined before the coroutines
+    receiver is known and the queries themselves don't specify a
+    result type that isn't possible when a coroutine provides a
+    receiver to a sender in a `co_await` expression. It should still
+    be possible to provide a user-customisable environment from the
+    receiver used by `co_await` expressions. One aspect of this
+    environment is to forward stop requests to `co_await`ed child
+    operations.
+7. The coroutine should be able to indicate that it was cancelled,
+    i.e., to get `set_stopped()` called on the task's receiver.
+    `std::execution::with_awaitable_senders` already provided this
+    ability senders being `co_await`ed but that doesn't necessarily
+    extend to the coroutine implementation.
+8. Similar to indicating that a task got cancelled it would be good
+    if a task could indicate that an error occurred without throwing
+    an exception which escapes from the coroutine.
+9. In general a task has to assume that an exception escapes a the
+    coroutine implementation. As a result, the task's completion
+    signatures need to include `set_error_t(std::exception_ptr)`.
+    If it can be indicated to the tas that no exception will escape
+    the coroutine, this completion signature can be avoided.
+10. When many `co_await`ed operations complete synchronously, there
+    is a chance for stack overflow. It may be reasonable to have
+    the implementation prevent that by using a suitable scheduler
+    sometimes.
+11. In some situations it can be useful to somehow schedule an
+    asychronous clean-up operation which is triggered upon
+    coroutine exit.
```

Comment:
This is true, but the approach taken by Folly ([`folly::coro::co_scope_exit`](https://github.com/facebook/folly/blob/main/folly/coro/ScopeExit.h#L357)) and Unifex ([`unifex::at_coroutine_exit`](https://github.com/facebookexperimental/libunifex/blob/main/include/unifex/at_coroutine_exit.hpp#L386)) probably shouldn't be emulated. The lifetimes are weird and confusing, and the rules around which references can and can't be captured are hard to learn. @snarkmaster has been working on an "async closure" in Folly that will replace `co_scope_exit` and I think it's a much better approach.